### ISSUE TITLE
Internet NEG backends must not have a health check

### ISSUE THREAD

**Issue by migueloller** [pulumi/pulumi-gcp#380]
Pulumi attempts to enforce the `healthChecks` field on the `compute.BackendService` resource but this field must not be passed to the GCP API in certain circumstances.

```typescript
// The following infrastructure is used so that we can use Cloud CDN with Cloud Functions.
const cloudFunctionsNEG = new gcp.compute.GlobalNetworkEndpointGroup(
  'cloud-functions',
  {
    networkEndpointType: 'INTERNET_FQDN_PORT',
  },
  { provider: gcpProvider },
)
const cloudFunctionsNE = new gcp.compute.GlobalNetworkEndpoint(
  'cloud-functions',
  {
    globalNetworkEndpointGroup: cloudFunctionsNEG.name,
    fqdn: `us-central1-${project}.cloudfunctions.net`,
    port: 443, // HTTPS
  },
  { provider: gcpProvider },
)
const cloudFunctionsBS = new gcp.compute.BackendService(
  'cloud-functions',
  // @ts-expect-error: `healthChecks` is required by Pulumi but backend services with
  // internet NEG backends must not have a health check according to GCP docs.
  //
  // See https://cloud.google.com/compute/docs/reference/rest/v1/backendServices
  {
    backends: [{ group: cloudFunctionsNEG.id }],
    enableCdn: true,
    protocol: 'HTTPS',
  },
  { provider: gcpProvider },
)
```

Unfortunately the `@ts-expect-error` directive doesn't help here since the SDK checks the field at runtime.

**Comment by leezen** [comment]
Looks like this is tracked upstream in https://github.com/terraform-google-modules/terraform-google-lb-http/issues/105 and once that's fixed should get picked up with the next pulumi-gcp update.

**Comment by lmakarov** [comment]
This has been resolved upstream.

**Comment by leezen** [comment]
@lmakarov Thanks for the friendly notification on this!

**Comment by lmakarov** [comment]
@leezen I've just updated to the latest version of pulumi (2.11.0), but I'm still getting this error:

```
  gcp:compute:BackendService (web-service):
    error: Error creating BackendService: googleapi: Error 400: Invalid value for field 'resource.healthChecks': ''. At least one health check needs to be specified., invalid
```

```js
let webService = new gcp.compute.BackendService("web-service", {
  protocol: "HTTPS",
  backends: [
      {
          group: webNEG.selfLink,
      },
  ],
});
```

I assume, a new release of pulumi/pulumi-gcp is necessary to address this issue?

**Comment by leezen** [comment]
@lmakarov This is dependent on the version of `@pulumi/gcp`

**Comment by leezen** [comment]
It looks like I referenced the wrong issue actually earlier and should have referenced https://github.com/hashicorp/terraform-provider-google/issues/6155 which is fixed in 3.32 of the TF provider, which corresponds to the `3.16.0` update (and subsequent) of the `@pulumi/gcp` package.

**Comment by lmakarov** [comment]
> which is fixed in 3.32 of the TF provider, which corresponds to the 3.16.0 update (and subsequent) of the @pulumi/gcp package.

@leezen The example error I posted earlier was with`@pulumi/gcp` v3.25.0. Any debug output I can provide to help diagnose the issue?

**Comment by lmakarov** [comment]
Never mind. Figured out the issue. 
I was mistakenly using `gcp.compute.NetworkEndpointGroup` instead of `gcp.compute.GlobalNetworkEndpointGroup`.

The following works as expeceted:

```js
const webNEG = new gcp.compute.GlobalNetworkEndpointGroup("web-neg", {
  networkEndpointType: "INTERNET_FQDN_PORT",
});

const webService = new gcp.compute.BackendService("web-service", {
  protocol: "HTTPS",
  backends: [
      {
        group: webNEG.selfLink,
      },
  ],
});

```
